packages/email-petition/src/email-petition.tsx

```
@@ -256,6 +313,58 @@ export function EmailPetition(props: {
 
   const recaptchaRef = useRef<ReCAPTCHA>(null);
 
+  // Function to call zipToCityLookup API
+  const fetchCityByZip = async (zipCode: string) => {
+    // Don't make API calls if Google Maps API is disabled
+    if (props.citySelectionMode !== "autocompleteTextbox") {
+      return;
+    }
+
+    if (!zipCode || zipCode.length !== 5) {
+      return;
```

pls throw an error here instead instead of failing silently.

---

this isn't neccessarily an error. This state is reached if zipcode textbox is empty or !==5. If I were to throw an error, devtools would keep throwing errors every time I changed the zipcode.

---

perhaps you are thinking of the useEffect code? which is already checking that the length is 5 before calling fetchCityByZip.

```
    if (zip && !outsideUS && zip.length === 5) {
      if (props.citySelectionMode === "autocompleteTextbox") {
        fetchCityByZip(zip);
```

---

bump :)